### PR TITLE
17 contact us

### PR DESCRIPTION
**Pull request by Amoodaa**
**Issue**: #17
## What did I do?

- [ ] Implemented the design
- [ ] Controlled the form with state
- [ ] Used netlify forms to submit the form data

## How to test?
`yarn start` into localhost:8000/contact-us

### CODE REVIEW

File: src/pages/contact-us/index.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+const CHANGE = 'CHANGE'
+
+const fields = [
+  { name: 'name', label: 'Full name' },
+  { name: 'organization', label: 'Company or organisation' },
+  { name: 'email', type: 'email', label: 'Email address' },
+  { name: 'phone-number', type: 'tel', label: 'Phone number' },
+]
+
+const reducer = (s, action) => {
```
Comment (yosefanajjar) [inline src/pages/contact-us/index.js:24]: What is s?

Comment (yosefanajjar) [reply src/pages/contact-us/index.js:24]: I know it's the state but come on it took me a minute to understand that

Comment (yosefanajjar) [reply src/pages/contact-us/index.js:24]: if eslint is giving you errors name it in a different understandable way or name this state and change the other one to formData for example

File: src/pages/contact-us/index.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    phone: '',
+  })
+
+  return (
+    <form
+      method="post"
+      name="contact-us"
+      action="/contact-us/success"
+      netlify
+      netlify-honeypot="bot-field"
```
Comment (yosefanajjar) [inline src/pages/contact-us/index.js:48]: Can you link the part of the docs that is referencing this?

File: src/pages/contact-us/index.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        onChange={({ target: { value } }) =>
+          dispatch({ type: CHANGE, key: 'message', value })
+        }
+      />
+      <br />
+      <br />
+      <br />
+      <br />
+      <br />
+      <br />
```
Comment (yosefanajjar) [inline src/pages/contact-us/index.js:84]: Can't you achieve the same result using margins?

File: src/pages/contact-us/index.js (hunk 2 above)
Comment (Amoodaa) [reply src/pages/contact-us/index.js:48]: https://docs.netlify.com/forms/setup/

File: src/pages/contact-us/index.js (hunk 1 above)
Comment (Amoodaa) [reply src/pages/contact-us/index.js:24]: its a state in a reducer.. its naturally for state

**Comment by yosefanajjar** [comment]
Where is your validation for the inputs? I have tried sending a text as a phone number and it worked + I don't think the default form validation looks great

File: src/pages/contact-us/index.js (hunk 3 above)
Comment (Amoodaa) [reply src/pages/contact-us/index.js:84]: felt like this will look better on different screen sizes

**Comment by yosefanajjar** [comment]
Screenshot from 2020-05-27 17-41-32

fix this bug in the style

File: src/pages/contact-us/index.js (hunk 1 above)
Comment (yosefanajjar) [reply src/pages/contact-us/index.js:24]: hmmm we know that but using one letter for your variable is not a sign for readable code, it is a suggestion after all

File: src/pages/contact-us/index.js (hunk 3 above)
Comment (yosefanajjar) [reply src/pages/contact-us/index.js:84]: Use media queries, your code and styles will look better

**Pull request merged**